[GUI] "gui.set_image" no longer accepts multi-channel scalar tensors

Source: https://forum.taichi.graphics/t/homework0-barnsleyfern/804/2

This seems to be introduced recently between v0.6.7 and v0.6.10.

**To Reproduce**
```python
import taichi as ti
import math

ti.init(arch=ti.gpu)

width = 660
height = 1000
pixels = ti.var(dt=ti.f32, shape=(width, height, 3))

gui = ti.GUI("test", (width, height))
gui.set_image(pixels)
```

Output:
```
AssertionError: Image resolution does not match GUI resolution
```

--- a/docs/index.rst
+++ b/docs/index.rst
@@ -62,7 +62,7 @@ The Taichi Programming Language
    :caption: Miscellaneous
    :maxdepth: 3
 
-   external_libraries
+   extension_libraries
    gui
    export_results
    cli_utilities

--- a/python/taichi/misc/gui.py
+++ b/python/taichi/misc/gui.py
@@ -83,20 +83,22 @@ class GUI:
         import taichi as ti
 
         if isinstance(img, ti.Expr):
-            if ti.core.is_integral(img.data_type()):
-                # image of uint is not optimized by xxx_to_image
-                self.img = self.cook_image(img.to_numpy())
-            else:
+            if ti.core.is_real(img.data_type()) and len(img.shape()) == 2:
+                # Type matched! We can use an optimized copy kernel.
                 assert img.shape(
                 ) == self.res, "Image resolution does not match GUI resolution"
                 from taichi.lang.meta import tensor_to_image
                 tensor_to_image(img, self.img)
                 ti.sync()
+            else:
+                # image of uint is not optimized by xxx_to_image
+                self.img = self.cook_image(img.to_numpy())
 
         elif isinstance(img, ti.Matrix):
             if ti.core.is_integral(img.data_type()):
                 self.img = self.cook_image(img.to_numpy())
             else:
+                # Type matched! We can use an optimized copy kernel.
                 assert img.shape(
                 ) == self.res, "Image resolution does not match GUI resolution"
                 assert img.n in [

--- a/python/taichi/misc/gui.py
+++ b/python/taichi/misc/gui.py
@@ -83,16 +83,16 @@ class GUI:
         import taichi as ti
 
         if isinstance(img, ti.Expr):
-            if ti.core.is_real(img.data_type()) and len(img.shape()) == 2:
+            if ti.core.is_integral(img.data_type()) or len(img.shape()) != 2:
+                # Images of uint is not optimized by xxx_to_image
+                self.img = self.cook_image(img.to_numpy())
+            else:
                 # Type matched! We can use an optimized copy kernel.
                 assert img.shape(
                 ) == self.res, "Image resolution does not match GUI resolution"
                 from taichi.lang.meta import tensor_to_image
                 tensor_to_image(img, self.img)
                 ti.sync()
-            else:
-                # image of uint is not optimized by xxx_to_image
-                self.img = self.cook_image(img.to_numpy())
 
         elif isinstance(img, ti.Matrix):
             if ti.core.is_integral(img.data_type()):

